Allow building on Fedora

3mkusiak opened the pull request:
This patch adds necessary modificatons for building and flashing the demo under Fedora. Some of the tasks need to be executed under the container.

dancesWithMachines committed:
Allow building on Fedora

This patch adds necessary modificatons for building and flashing the
demo under Fedora. Some of the tasks need to be executed under the
container.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
README: Add info on workaround

Add info how to run demos on A3 revision of 55S69 platform. Reorder
readme to ensure users apply workarounds and env quirks first.

Related issues:
https://github.com/crosscon/UC1.1-Manifest/issues/2#issuecomment-3027862206
https://github.com/crosscon/CROSSCON-Hypervisor-and-TEE-Isolation-Demos/issues/30#issuecomment-3027875435

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

DaniilKl commented:
@3mkusiak, why using a specific version of `tio` and not any other tools for serial communication?

DaniilKl commented on README.md:
```diff
@@ -60,6 +46,65 @@ And to fetch required code for all zephyr targets run:
+about the issue
+[here](https://github.com/crosscon/UC1.1-Manifest/issues/2#issuecomment-3027862206).
+
+A workaround is to checkout to '0.2.1` revision which does not include
```
```suggestion
A workaround is to checkout to `0.2.1` revision which does not include
```

DaniilKl commented on scripts/install_tio.sh:
```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+set -e
```
With `e` set this script will fail on every `return` other then `0` and all the functions where you use the `return` will not work. E.g. running script with no `tio` installed (I added `set -x` for tracing):

```bash
λ scripts/install_tio.sh    
+ REQUIRED_TIO_VERSION=3.8
+ TIO_SOURCE_URL=https://github.com/tio/tio/releases/download/v3.8/tio-3.8.tar.xz
++ tio_version
++ command -v tio
++ echo ''
++ return 1
+ current_version=
```

The script stopped just after returning `1` from `tio_version`.

3mkusiak replied:
[Fixed](https://github.com/3mdeb/UC1.1-Manifest/pull/1/commits/c2eb146951a09f1fb6b979c57af2dba412703d1e#diff-55d72764df415a31d9781ffd42dd1673b178d8efbce31afb6b9dd9b59d6f1541)

ghost commented:
> @3mkusiak, why using a specific version of tio and not any other tools for serial communication?

Because [this certain version is required](https://github.com/crosscon/UC1.1-Manifest/blob/main/README.md?plain=1#L16).  The devs at Crosscon should be asked this. The `tio` is used here to fetch output from enrollment app and automatically parse it, it is a part of automation, cannot be simply replaced.

dancesWithMachines committed:
scripts: Fix install_tio

Fix install_tio.sh script. The issue is due to "set -e" when functions
returned non zero status, the exectuion hanged. Switch returning codes
to returning stings.

Fix formatting issue in README.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

DaniilKl commented:
I do not have other comments here. I cannot resolve above threads probably because of access rights.

DaniilKl commented:
Ok, now I have the access.
